x/auth/tx/xauthclient.go

```diff
@@ -0,0 +1,159 @@
+// Package tx 's xauthclient.go file is copy-pasted from
+// https://github.com/cosmos/cosmos-sdk/blob/v0.41.3/x/auth/client/query.go
+// It is duplicated as to not introduce any breaking change in 0.41.4, see PR:
+// TODO
+// It is refactored and removed in the following PR:
+// TODO
```

Comment:
This PR fixes query Txs (by hash or by event) GRPC handlers, by filling out all required fields.

The thing is, all this logic is already implemented in `x/auth/client/query.go`, so the GRPC handlers could just call that file. Unfortunately, that creates an import cycle.

To not introduce breaking changes, I copy-pasted the relevant functions from `x/auth/client/query.go` to this folder. We have **duplicate code**, and will have it on 0.41.4.

To remove duplication, I created https://github.com/cosmos/cosmos-sdk/pull/8734, which will be on master and on 0.42+.

Reply:
TBH: I'm also okay to just merge #8734 into this PR, so that we only have one clean PR. The downside is that it introduces an API-breaking change in a nested package.